### DISCUSSION TITLE
Show text with after 3 sec with setTimeOut in React js

### DISCUSSION THREAD

**Discussion by lookatlion** [community/community#53813] [Programming Help] [answered]
### Select Topic Area

Question

### Select Feature Area

- [X] Issues
- [ ] Projects

### Body

For React JS I need to show a text in

after 2 seconds, it's work in console but not in the DOM

  const items = document.getElementById("items");
  const errorDisplay = () => {
    setTimeout(function () {
      items.innerHTML = "please wait..";
      console.log(items);
    }, 2000);
  };
  <p id="items">  </p>

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by iamsuiux** [comment, marked as answer]
If you want to show a please wait paragraph for two seconds before displaying the products you can do something like this:

import { useEffect, useState } from "react";

const items = [{ id: 1 }, { id: 2 }];

export default function App() {
const [isWaiting, setIsWaiting] = useState(true);

const productsListItems = items.map((item) => <li>{item.id}</li>);

useEffect(() => {
const timer = setTimeout(() => {
  setIsWaiting(false);

  return () => clearTimeout(timer);
 }, [2000]);
}, []);

return (
<div className="App">
  {isWaiting && <p>Please wait...</p>}
  {!isWaiting && <ul>{productsListItems}</ul>}
</div>
 );
}
